src/main/java/org/swisspush/redisques/QueueStatsService.java

```diff
@@ -0,0 +1,219 @@
+    private <CTX> void fetchQueueNamesAndSize(GetQueueStatsRequest<CTX> req, BiConsumer<Throwable, GetQueueStatsRequest<CTX>> onDone) {
+        String filter = req.mentor.filter(req.mCtx);
+        JsonObject operation = buildGetQueuesItemsCountOperation(filter);
+        eventBus.<JsonObject>request(redisquesAddress, operation, ev -> {
+            if (ev.failed()) {
+                onDone.accept(new Exception("eventBus.request()", ev.cause()), null);
+                return;
+            }
+            Message<JsonObject> msg = ev.result();
+            JsonObject body = msg.body();
+            String status = body.getString(STATUS);
+            if (!OK.equals(status)) {
+                onDone.accept(new Exception("Unexpected status " + status), null);
+                return;
+            }
+            JsonArray queuesJsonArr = body.getJsonArray(QUEUES);
+            if (queuesJsonArr == null || queuesJsonArr.isEmpty()) {
+                log.debug("result was {}, we return an empty result.", queuesJsonArr == null ? "null" : "empty");
+                req.queues = emptyList();
+                onDone.accept(null, req);
+                return;
+            }
+            boolean includeEmptyQueues = req.mentor.includeEmptyQueues(req.mCtx);
+            List<Queue> queues = new ArrayList<>(queuesJsonArr.size());
+            for (var it = queuesJsonArr.iterator(); it.hasNext(); ) {
+                JsonObject queueJson = (JsonObject) it.next();
+                String name = queueJson.getString(MONITOR_QUEUE_NAME);
+                Long size = queueJson.getLong(MONITOR_QUEUE_SIZE);
+                // No need to process empty queues any further if caller is not interested
+                // in them anyway.
+                if (!includeEmptyQueues && (size == null || size == 0)) continue;
+                Queue queue = new Queue();
+                queue.name = name;
+                queue.size = size;
+                queues.add(queue);
+            }
+            queues.sort(this::compareLargestFirst);
+            // Only the part with the most filled queues got requested. Get rid of
+            // all shorter queues then.
+            int limit = req.mentor.limit(req.mCtx);
+            if (limit != 0 && queues.size() > limit) queues = queues.subList(0, limit);
+            req.queues = queues;
+            onDone.accept(null, req);
+        });
+    }
+
+    private <CTX> void fetchRetryDetails(GetQueueStatsRequest<CTX> req, BiConsumer<Throwable, GetQueueStatsRequest<CTX>> onDone) {
+        long begGetQueueStatsMs = currentTimeMillis();
+        assert req.queueNames != null;
+        queueStatisticsCollector.getQueueStatistics(req.queueNames).onComplete( ev -> {
+            req.queueNames = null; // <- no longer needed
+            long durGetQueueStatsMs = currentTimeMillis() - begGetQueueStatsMs;
+            if (durGetQueueStatsMs > 42) log.debug("queueStatisticsCollector.getQueueStatistics() took {}ms", durGetQueueStatsMs);
```

Comment:
Is there a reason for not logging it to _debug_ if it was fast enough? And is 42 a good cutoff point or is it just funny? :-)

Reply:
No idea :D WDYT should we just log unconditionally?

Reply:
If you have a _good_ reason _not_ to log it below some threshold, then please do. If the condition is more or less random, then remove it. After all `log.debug()` is already logging _conditionally_ based on log level.